spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -436,6 +436,9 @@ public SubscriptionAdminClient subscriptionAdminClient(
   @ConditionalOnMissingBean(name = "subscriberTransportChannelProvider")
   public TransportChannelProvider subscriberTransportChannelProvider() {
     return SubscriberStubSettings.defaultGrpcTransportProviderBuilder()
+        // default value specified by pubsub client library,
+        // see https://github.com/googleapis/java-pubsub/blob/main/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java#L487.
+        .setMaxInboundMetadataSize(4 * 1024 * 1024)
```

Comment:
Is it possible to reference the default variable here instead of hardcoding the default value? Just so that we are aligned with the default in the client in case it changes in the future.

Reply:
Thanks for checking, That is a little unfortunate that it can't be referenced. It might be worth creating a ticket in the client repo (similar to https://github.com/googleapis/java-pubsub/issues/373)

Reply:
I think the builder is autogenerated by the generator: https://github.com/googleapis/java-pubsub/blob/main/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/SubscriptionAdminSettings.java#L246 and https://github.com/googleapis/java-pubsub/blob/main/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/stub/SubscriberStubSettings.java#L425